File: problems/chapter06/miiitomi/006.cpp
// AtCoder ABC026   D - 高橋君ボール1号
// https://atcoder.jp/contests/abc026/tasks/abc026_d

#include <bits/stdc++.h>
using namespace std;

int A, B, C;

double f(double t) {
    return (A * t) + B * sin(C * t * M_PI);
}

int main() {
    cin >> A >> B >> C;
    double left = (100.0 - B) / A;
    double right = (100.0 + B) / A;
    double mid = (left + right) / 2;

    while (abs(f(mid) - 100.0) >= 1e-7) {
        if (f(mid) > 100.0) {
            right = mid;
        } else {
            left = mid;
        }
        mid = (left + right) / 2;
    }

    cout << fixed << setprecision(10) << mid << endl;
}

File: problems/chapter06/mshibatatt/006.py
# https://atcoder.jp/contests/abc026/tasks/abc026_d

from math import sin, pi

def main():
    a, b, c = map(int, input().split())

    left = 0
    right = (100+b)/a + 1
    t = left + (right - left)/2
    ft = a*t + b*sin(c*t*pi) - 100
    while abs(ft) > 10**(-6):
        if ft > 0:
            right = t
        else:
            left = t
        t = left + (right - left)/2
        ft = a*t + b*sin(c*t*pi) - 100
        
    print(t)

if __name__=='__main__':
    main()